src/recipe/views.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import viewsets, status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from core.models import Recipe, Ingredient
+from recipe import serializers
+
+
+class IngredientViewSet(viewsets.ModelViewSet):
+    """Manage ingredients in the database"""
+    serializer_class = serializers.IngredientSerializer
+    queryset = Ingredient.objects.all()
+
+
+class RecipeViewSet(viewsets.ModelViewSet):
+    """Manage recipes in the database"""
+    serializer_class = serializers.RecipeSerializer
+    queryset = Recipe.objects.all()
+
+    def get_serializer_class(self):
+        """Return appropriate serializer class"""
+        if self.action == 'add_ingredient':
+            return serializers.IngredientSerializer
+        return self.serializer_class
+
+    @action(methods=['POST'], detail=True, url_path='add-ingredient')
```

Comment:
Did you try to use one of django restframework's mixins?
https://www.django-rest-framework.org/api-guide/generic-views/#createmodelmixin

Reply:
Never thought about it. `@action` was part of the course and use it. I gave it some time boxing probably a cleaner approach in terms of API would be using a nested router https://github.com/alanjds/drf-nested-routers but using an extra dependency was a no go for me.

How would a mixing help in this situation?